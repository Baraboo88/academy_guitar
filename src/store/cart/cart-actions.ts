import {CartItemModel, GuitarModel} from '../../types/guitar-model';

export enum CartAction {
  AddOneToCartItems = 'cart/add-one-to-cart-items',
  AddCustomToCartItems = 'cart/add-custom-to-cart-items',
  DeleteOneFromCartItems = 'cart/delete-one-from-cart-items',
  DeleteFromCartItems ='cart/delete-from-cart-items',
  SetDiscountPercent ='cart/set-discount-percent',
  SetErrorMessage = 'cart/set-error-message',
  SetIsResponseReceived = 'cart/set-is-response-received'
}

export const CartActionCreator = {
  addOneToCartItems(guitar: GuitarModel) {
    return {type: CartAction.AddOneToCartItems, payload: guitar};
  },
  addCustomToCartItems(guitarWithCount: CartItemModel) {
    return {type: CartAction.AddCustomToCartItems, payload: guitarWithCount};
  },
  deleteOneFromCartItems(guitar: GuitarModel) {
    return {type: CartAction.DeleteOneFromCartItems, payload: guitar};
  },
  deleteFromCartItems(guitar: GuitarModel) {
    return {type: CartAction.DeleteFromCartItems, payload: guitar};
  },
  setDiscountPercent(discount: number) {
    return {type: CartAction.SetDiscountPercent, payload: discount};
  },
  setErrorMessage(error: string) {
    return {type: CartAction.SetErrorMessage, payload: error};
  },
  setIsResponseReceived(isResponseReceived: boolean) {
    return {type: CartAction.SetIsResponseReceived, payload: isResponseReceived};
  },
};
