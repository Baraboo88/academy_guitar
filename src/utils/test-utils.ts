import {AddCommentModel, GuitarCommentModel, GuitarModel, GuitarType} from '../types/guitar-model';
import {createApi} from '../api';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {ReactWrapper} from 'enzyme';


export const MOCK_ERROR= 'Somme error';

export const mockGuitars: GuitarModel[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: 'electric' as GuitarType,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 4,
    price: 17500,
  },
  {
    id: 2,
    name: 'CURT Z300',
    vendorCode: 'TK129049',
    type: 'electric' as GuitarType,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  },
  {
    id: 3,
    name: 'Roman LX',
    vendorCode: 'RO111111',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле класса премиум от компании CURT, собравшая в себе все самые необходимые качесва: лёгкость корпуса, прочность струн и компактный размер.',
    previewImg: 'img/guitar-6.jpg',
    stringCount: 4,
    rating: 4,
    price: 6800,
  },
  {
    id: 4,
    name: 'CURT T300',
    vendorCode: 'TK436457',
    type: 'electric' as GuitarType,
    description: 'CURT T300 - это шестиструнная электрогитара популярной линейки FPT. Модель с классическим стилем головы грифа и деки.',
    previewImg: 'img/guitar-3.jpg',
    stringCount: 6,
    rating: 5,
    price: 30000,
  },
  {
    id: 5,
    name: 'Dania Super',
    vendorCode: 'DI192138',
    type: 'acoustic' as GuitarType,
    description: 'Гитары производителя Dania пользуются популярностью у музыкантов разного уровня. В модели Super идеально сочетаются демократичная цена и качество. Корпус, выполненный из тополя позволяет корпусу долгое время сохранять свой первоначальный вид.',
    previewImg: 'img/guitar-4.jpg',
    stringCount: 7,
    rating: 4.5,
    price: 3500,
  },
  {
    id: 6,
    name: 'Честер WX',
    vendorCode: 'SO934345',
    type: 'electric' as GuitarType,
    description: 'Электрогитара с олд-скульным грифом и глянцевым корпусом для настоящих рок-музыкантов. Насыщенный звук идеально подойдет для исполнения рок-композиций.',
    previewImg: 'img/guitar-2.jpg',
    stringCount: 6,
    rating: 3.5,
    price: 15300,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
export const testInitialState = {
  currentGuitar: {
    currentGuitar: mockGuitars,
    errorMsg: '',
  },
  guitars: {
    guitars: mockGuitars,
    errorMsg: '',
  },
};

export const mockCommentToAdd: AddCommentModel = {
  advantage: 'Mock Advantage', comment: 'Mock Comment', disadvantage: 'Mock Disadvantage', guitarId: 3, rating: 3, userName: 'Some name',
};


export const mockComment: GuitarCommentModel = {id: '31b10247-adff-44a7-937e-c71af64f425b',
  userName: 'Никита',
  advantage: 'Цена.',
  disadvantage: 'Быстро приходит в негодность покрытие краской.',
  comment: 'Хорошая гитара для начинающих, сделана качественно. Лучше за эти деньги не найти.',
  rating: 5,
  createAt: new Date('2022-03-10T10:56:01.590Z'),
  guitarId: 1};

export const getTestStore = () => {

  const reducer = (state = testInitialState) => state;
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};

export const findByTestAtr = (component:ReactWrapper, attr: string) => component.find(`[data-test='${attr}']`);
