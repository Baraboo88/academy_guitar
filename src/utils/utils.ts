const STARTS_AMOUNT = 5;

export const imageAdapter = (img: string) => img.slice(img.length - 5, img.length - 4);


export const generateStarsArray = () => {
  const stars = [];
  for (let i = 1; i <= STARTS_AMOUNT; i++) {
    stars.push(i);
  }
  return stars;
};

export const getCyrillicRating = (rating: number) => {
  if(rating> 0 && rating < 3){
    return 'Плохо';
  } else if (rating >= 3 && rating < 4){
    return 'Удовлетворительно';
  } else if(rating >= 4 && rating < 5){
    return 'Хорошо';
  } else {
    return 'Отлично';
  }
};
