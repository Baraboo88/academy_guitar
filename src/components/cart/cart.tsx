import React, {ChangeEvent, useEffect, useState} from 'react';
import Header from '../header/header';
import Footer from '../footer/footer';
import {StateModel} from '../../types/redux-models';

import {ThunkDispatch} from 'redux-thunk';
import {RootState} from '../../index';
import {AxiosStatic} from 'axios';
import {Action} from 'redux';

import {connect} from 'react-redux';

import {CartItemModel, GuitarModel} from '../../types/guitar-model';
import {
  getAdapterImage,
  getCyrillicType,
  getPriceWithSpaces, INITIAL_CART_ITEM_COUNT
} from '../../utils/utils';
import {Link} from 'react-router-dom';
import {
  getCartErrorMessage, getCartIsResponseReceived,
  getCartItems, getCartItemsDiscountAmount,
  getCartItemsTotalAmount
} from '../../store/cart/cart-selector';
import {CartActionCreator, CartOperation} from '../../store/cart/cart-actions';
import ConfirmCartItemDeleteModal from '../confirm-cart-item-delete-modal/confirm-cart-item-delete-modal';
import {TailSpin} from 'react-loader-spinner';


interface CartProps{
  cartItems: CartItemModel [];
  addOneToCartItems: (guitar: GuitarModel) => void;
  addCustomToCartItems: (guitarWithCount: CartItemModel) => void;
  deleteOneFromCartItems:(guitar: GuitarModel) => void;
  deleteFromCartItems: (guitar: GuitarModel) => void;
  discountAmount: number;
  errorMessage: string;
  isResponseReceived: boolean;
  getPromoDiscount: (promo: string) => void;
  resetCartErrorMessage: () => void;
  totalAmount: number
}

function Cart(props: CartProps) {
  const {cartItems,
    addOneToCartItems,
    addCustomToCartItems,
    deleteOneFromCartItems,
    deleteFromCartItems,
    discountAmount,
    errorMessage,
    getPromoDiscount,
    resetCartErrorMessage,
    totalAmount,
    isResponseReceived,
  } = props;

  const[guitarToDelete, setGuitarToDelete] = useState<GuitarModel | null>(null);
  const [promo, setPromo] = useState('');
  const [isPromoInvalid, setIsPromoInvalid] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [innerCartItems, setInnerCartItems] = useState<CartItemModel []>([]);

  useEffect(() => {
    setInnerCartItems(cartItems);
  }, [cartItems]);

  useEffect(() => {
    if(isResponseReceived && isLoading){
      setIsLoading(false);
    }
  }, [isResponseReceived, isLoading]);

  useEffect(() => {
    if(errorMessage){
      setIsPromoInvalid(true);
    } else {
      setIsPromoInvalid(false);
    }
  }, [errorMessage]);


  const handlerCartItemDelete = () => {
    if(guitarToDelete){
      deleteFromCartItems(guitarToDelete);
      setGuitarToDelete(null);
    }
  };

  const handlerConfirmDeleteModalClose = () => {
    setGuitarToDelete(null);
  };

  const handlerInnerCartItemIncrease = (guitar: GuitarModel) => {
    addOneToCartItems(guitar);
  };

  const handlerGuitarCountChange = (guitar: GuitarModel, count: number) => {

    const newInnerCartItems = innerCartItems.map((cartItem) => {
      const isChangedItem = cartItem.guitar.id === guitar.id;
      if(isChangedItem){
        const newCartItem = {...cartItem};
        newCartItem.count = count;
        return newCartItem;
      }
      return cartItem;
    });

    setInnerCartItems(newInnerCartItems);
  };

  const handlerGuitarCountBlur = (guitar:GuitarModel) => {

    const changedItem = innerCartItems.find((item) => item.guitar.id === guitar.id);

    if(changedItem){
      if(changedItem.count <= 0 ){
        setGuitarToDelete(guitar);
      } else {
        addCustomToCartItems({guitar, count: changedItem.count });
      }
    }

  };


  const handlerCartItemDecrease = (cartItem:CartItemModel) => {
    if(cartItem.count === INITIAL_CART_ITEM_COUNT){
      setGuitarToDelete(cartItem.guitar);
    } else {
      deleteOneFromCartItems(cartItem.guitar);
    }
  };

  const handlerPromoCodeSubmit = (evt:React.SyntheticEvent) => {
    evt.preventDefault();
    setIsLoading(true);
    getPromoDiscount(promo.trim());
  };

  const handlerPromoChange = (evt: ChangeEvent<HTMLInputElement>) => {
    setPromo(evt.target.value);
    resetCartErrorMessage();
  };

  const renderCartItems  = () => innerCartItems.map((cartItem: CartItemModel) =>    (
    <div className="cart-item" key={cartItem.guitar.id}>
      <button onClick={() => {
        setGuitarToDelete(cartItem.guitar);
      }} className="cart-item__close-button button-cross" type="button" aria-label="Удалить"
      >
        <span
          className="button-cross__icon"
        >
        </span><span className="cart-item__close-button-interactive-area"></span>
      </button>
      <div className="cart-item__image">
        <img  src={`/img/content/catalog-product-${getAdapterImage(cartItem.guitar.previewImg)}.jpg`}
          srcSet={`/img/content/catalog-product-${getAdapterImage(cartItem.guitar.previewImg)}@2x.jpg 2x`} width="55"
          height="130" alt={cartItem.guitar.name}
        />
      </div>
      <div className="product-info cart-item__info">
        <p className="product-info__title">{cartItem.guitar.name}</p>
        <p className="product-info__info">Артикул: {cartItem.guitar.vendorCode}</p>
        <p className="product-info__info">{getCyrillicType(cartItem.guitar.type)}, {cartItem.guitar.stringCount} струнная</p>
      </div>
      <div className="cart-item__price">{getPriceWithSpaces(cartItem.guitar.price)} ₽</div>
      <div className="quantity cart-item__quantity">
        <button data-test ='test-decrease-one' onClick={() => {
          handlerCartItemDecrease(cartItem);
        }} className="quantity__button" aria-label="Уменьшить количество"
        >
          <svg width="8" height="8" aria-hidden="true">
            <use xlinkHref="#icon-minus"></use>
          </svg>
        </button>
        <input  data-test ='test-custom-count' onBlur={() => {
          handlerGuitarCountBlur(cartItem.guitar);
        }} onChange={(evt) => {
          handlerGuitarCountChange(cartItem.guitar, Number(evt.target.value));
        }} value={cartItem.count === 0 ? '' : cartItem.count} className="quantity__input" type="number" placeholder={cartItem.count.toString()} id="2-count" name="2-count" max="99"
        />
        <button data-test ='test-add-one' onClick={() => {
          handlerInnerCartItemIncrease(cartItem.guitar);
        }}  className="quantity__button" aria-label="Увеличить количество"
        >
          <svg width="8" height="8" aria-hidden="true">
            <use xlinkHref="#icon-plus"></use>
          </svg>
        </button>
      </div>
      <div className="cart-item__price-total">{getPriceWithSpaces(cartItem.guitar.price * cartItem.count)} ₽</div>
    </div>));

  return (
    <div  className="wrapper">
      <Header />
      <main className="page-content">

        <div className="container">
          <h1 className="title title--bigger page-content__title">Корзина</h1>
          <ul className="breadcrumbs page-content__breadcrumbs page-content__breadcrumbs--on-cart-page">
            <li className="breadcrumbs__item"><Link to={'/'} className="link">Главная</Link>
            </li>
            <li className="breadcrumbs__item"><Link to={'/'} className="link" >Каталог</Link>
            </li>
            <li className="breadcrumbs__item"><span className="link">Корзина</span>
            </li>
          </ul>
          <div className="cart">
            {renderCartItems()}
            <div className="cart__footer">
              <div className="cart__coupon coupon">
                <h2 className="title title--little coupon__title">Промокод на скидку</h2>
                <p className="coupon__info">Введите свой промокод, если он у вас есть.</p>
                <form  data-test='test-promo-submit' className="coupon__form" id="coupon-form" method="post" action="/" onSubmit={handlerPromoCodeSubmit}>
                  <div className="form-input coupon__input">
                    <label className="visually-hidden">Промокод</label>
                    <input data-test ='test-change-promo' onChange={handlerPromoChange} type="text" placeholder="Введите промокод" id="coupon" name="coupon"/>
                    {discountAmount ? <p className="form-input__message form-input__message--success">Промокод принят</p> : ''}
                    {isPromoInvalid &&
                      <p className="form-input__message form-input__message--error">неверный промокод</p>}
                    {isLoading &&
                        <TailSpin  color="#000000" height={80} width={80} />}
                  </div>
                  <button className="button button--big coupon__button">Применить</button>
                </form>
              </div>
              <div className="cart__total-info">
                <p className="cart__total-item">
                  <span className="cart__total-value-name">Всего:</span>
                  <span
                    className="cart__total-value"
                  >{getPriceWithSpaces(totalAmount)} ₽
                  </span>
                </p>
                <p className="cart__total-item">
                  <span className="cart__total-value-name">Скидка:</span>
                  <span
                    className={`cart__total-value ${discountAmount > 0  ? 'cart__total-value--bonus' : ''}`}
                  >{discountAmount > 0 ? `- ${getPriceWithSpaces(discountAmount)}`: 0} ₽
                  </span>
                </p>
                <p className="cart__total-item">
                  <span className="cart__total-value-name">К оплате:</span>
                  <span
                    className="cart__total-value cart__total-value--payment"
                  >{getPriceWithSpaces(totalAmount - (discountAmount ? discountAmount: 0))} ₽
                  </span>
                </p>
                <button className="button button--red button--big cart__order-button">Оформить заказ</button>
              </div>
            </div>
          </div>
          {guitarToDelete !== null && <ConfirmCartItemDeleteModal guitar={guitarToDelete} onCartItemDelete={handlerCartItemDelete} onModalClose={handlerConfirmDeleteModalClose}/>}
        </div>

      </main>
      <Footer/>
    </div>
  );
}


const mapStateToProps = (state: StateModel) => ({
  cartItems: getCartItems(state),
  totalAmount: getCartItemsTotalAmount(state),
  discountAmount: getCartItemsDiscountAmount(state),
  errorMessage: getCartErrorMessage(state),
  isResponseReceived: getCartIsResponseReceived(state),
});


const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, AxiosStatic, Action>) => ({
  addOneToCartItems(guitar: GuitarModel) {
    dispatch(CartActionCreator.addOneToCartItems(guitar));
  },
  addCustomToCartItems(guitarWithCount: CartItemModel){
    dispatch(CartActionCreator.addCustomToCartItems(guitarWithCount));
  },
  deleteOneFromCartItems(guitar: GuitarModel){
    dispatch(CartActionCreator.deleteOneFromCartItems(guitar));
  },
  deleteFromCartItems(guitar: GuitarModel){
    dispatch(CartActionCreator.deleteFromCartItems(guitar));
  },
  getPromoDiscount(promoCode: string){
    dispatch(CartOperation.getPromoDiscount(promoCode));
  },
  resetCartErrorMessage(){
    dispatch(CartActionCreator.setErrorMessage(''));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
export {Cart};
