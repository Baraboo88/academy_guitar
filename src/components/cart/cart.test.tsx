import * as React from 'react';


import {Cart} from './cart';
import {getTestStore, mockCartItems} from '../../utils/test-utils';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';

import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({adapter: new EnzymeReactAdapter()});


it('Cart successfully rendered', () => {
  const mockSetAddOneToCartItems = jest.fn();
  const mockSetAddCustomToCartItems = jest.fn();
  const mockSetDeleteOneFromCartItems = jest.fn();
  const mockSetDeleteFromCartItems = jest.fn();
  const mockResetErrorMessage = jest.fn();
  const mockGetPromoDiscount = jest.fn();
  const tree = mount(
    <Provider store={getTestStore()}>
      <BrowserRouter>
        <Cart
          cartItems={mockCartItems}
          addOneToCartItems={mockSetAddOneToCartItems}
          addCustomToCartItems={mockSetAddCustomToCartItems}
          deleteOneFromCartItems={mockSetDeleteOneFromCartItems}
          deleteFromCartItems={mockSetDeleteFromCartItems}
          totalAmount={2000}
          resetCartErrorMessage={mockResetErrorMessage}
          getPromoDiscount={mockGetPromoDiscount}
          errorMessage={''}
          discountAmount={15}
          isResponseReceived
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(toJson(tree, {mode: 'deep'})).toMatchSnapshot();
});
