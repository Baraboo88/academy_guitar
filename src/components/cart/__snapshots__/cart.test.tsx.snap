// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart successfully rendered 1`] = `
<div
  className="wrapper"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container header__wrapper"
    >
      <a
        className="header__logo logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="/img/svg/logo.svg"
          width="70"
        />
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li>
            <a
              className="link main-nav__link "
              href="/"
              onClick={[Function]}
            >
              Каталог
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              Где купить?
            </a>
          </li>
          <li>
            <a
              className="link main-nav__link"
              href="#"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="form-search"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <form
          className="form-search__form"
          id="form-search"
        >
          <button
            className="form-search__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="form-search__icon"
              height="15"
              width="14"
            >
              <use
                xlinkHref="#icon-search"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              Начать поиск
            </span>
          </button>
          <input
            autoComplete="off"
            className="form-search__input"
            data-test="test-search-name"
            id="search"
            onChange={[Function]}
            placeholder="что вы ищите?"
            type="text"
          />
          <label
            className="visually-hidden"
            htmlFor="search"
          >
            Поиск
          </label>
        </form>
        <ul
          className="form-search__select-list hidden"
        >
          <li
            className="form-search__select-item"
            key="1"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Честер Bass
          </li>
          <li
            className="form-search__select-item"
            key="2"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            CURT Z300
          </li>
          <li
            className="form-search__select-item"
            key="3"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Roman LX
          </li>
          <li
            className="form-search__select-item"
            key="4"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            CURT T300
          </li>
          <li
            className="form-search__select-item"
            key="5"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Dania Super
          </li>
          <li
            className="form-search__select-item"
            key="6"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Честер WX
          </li>
          <li
            className="form-search__select-item"
            key="7"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            Dania VX
          </li>
        </ul>
        <button
          className="form-search__reset"
          form="form-search"
          onClick={[Function]}
          type="reset"
        >
          <svg
            aria-hidden="true"
            className="form-search__icon"
            height="15"
            width="14"
          >
            <use
              xlinkHref="#icon-close"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            Сбросить поиск
          </span>
        </button>
      </div>
      <a
        aria-label="Корзина"
        className="header__cart-link"
        href="/cart"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="header__cart-icon"
          height="14"
          width="14"
        >
          <use
            xlinkHref="#icon-basket"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          Перейти в корзину
        </span>
        <span
          className="header__cart-count"
        >
          5
        </span>
      </a>
    </div>
  </header>
  <main
    className="page-content"
  >
    <div
      className="container"
    >
      <h1
        className="title title--bigger page-content__title"
      >
        Корзина
      </h1>
      <ul
        className="breadcrumbs page-content__breadcrumbs page-content__breadcrumbs--on-cart-page"
      >
        <li
          className="breadcrumbs__item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Каталог
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <span
            className="link"
          >
            Корзина
          </span>
        </li>
      </ul>
      <div
        className="cart"
      >
        <div
          className="cart-item"
          key="1"
        >
          <button
            aria-label="Удалить"
            className="cart-item__close-button button-cross"
            onClick={[Function]}
            type="button"
          >
            <span
              className="button-cross__icon"
            />
            <span
              className="cart-item__close-button-interactive-area"
            />
          </button>
          <div
            className="cart-item__image"
          >
            <img
              alt="ЭлектроГитара Честер bass"
              height="130"
              src="/img/content/catalog-product-1.jpg"
              srcSet="/img/content/catalog-product-1@2x.jpg 2x"
              width="55"
            />
          </div>
          <div
            className="product-info cart-item__info"
          >
            <p
              className="product-info__title"
            >
              Честер Bass
            </p>
            <p
              className="product-info__info"
            >
              Артикул: 
              SO757575
            </p>
            <p
              className="product-info__info"
            >
              Электрогитара
              , 
              7
               струнная
            </p>
          </div>
          <div
            className="cart-item__price"
          >
            17 500
             ₽
          </div>
          <div
            className="quantity cart-item__quantity"
          >
            <button
              aria-label="Уменьшить количество"
              className="quantity__button"
              data-test="test-decrease-one"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                height="8"
                width="8"
              >
                <use
                  xlinkHref="#icon-minus"
                />
              </svg>
            </button>
            <input
              className="quantity__input"
              data-test="test-custom-count"
              id="2-count"
              max="99"
              name="2-count"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="3"
              type="number"
              value={3}
            />
            <button
              aria-label="Увеличить количество"
              className="quantity__button"
              data-test="test-add-one"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                height="8"
                width="8"
              >
                <use
                  xlinkHref="#icon-plus"
                />
              </svg>
            </button>
          </div>
          <div
            className="cart-item__price-total"
          >
            52 500
             ₽
          </div>
        </div>
        <div
          className="cart-item"
          key="2"
        >
          <button
            aria-label="Удалить"
            className="cart-item__close-button button-cross"
            onClick={[Function]}
            type="button"
          >
            <span
              className="button-cross__icon"
            />
            <span
              className="cart-item__close-button-interactive-area"
            />
          </button>
          <div
            className="cart-item__image"
          >
            <img
              alt="ЭлектроГитара Честер bass"
              height="130"
              src="/img/content/catalog-product-8.jpg"
              srcSet="/img/content/catalog-product-8@2x.jpg 2x"
              width="55"
            />
          </div>
          <div
            className="product-info cart-item__info"
          >
            <p
              className="product-info__title"
            >
              CURT Z300
            </p>
            <p
              className="product-info__info"
            >
              Артикул: 
              TK129049
            </p>
            <p
              className="product-info__info"
            >
              Электрогитара
              , 
              7
               струнная
            </p>
          </div>
          <div
            className="cart-item__price"
          >
            29 500
             ₽
          </div>
          <div
            className="quantity cart-item__quantity"
          >
            <button
              aria-label="Уменьшить количество"
              className="quantity__button"
              data-test="test-decrease-one"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                height="8"
                width="8"
              >
                <use
                  xlinkHref="#icon-minus"
                />
              </svg>
            </button>
            <input
              className="quantity__input"
              data-test="test-custom-count"
              id="2-count"
              max="99"
              name="2-count"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="2"
              type="number"
              value={2}
            />
            <button
              aria-label="Увеличить количество"
              className="quantity__button"
              data-test="test-add-one"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                height="8"
                width="8"
              >
                <use
                  xlinkHref="#icon-plus"
                />
              </svg>
            </button>
          </div>
          <div
            className="cart-item__price-total"
          >
            59 000
             ₽
          </div>
        </div>
        <div
          className="cart__footer"
        >
          <div
            className="cart__coupon coupon"
          >
            <h2
              className="title title--little coupon__title"
            >
              Промокод на скидку
            </h2>
            <p
              className="coupon__info"
            >
              Введите свой промокод, если он у вас есть.
            </p>
            <form
              action="/"
              className="coupon__form"
              data-test="test-promo-submit"
              id="coupon-form"
              method="post"
              onSubmit={[Function]}
            >
              <div
                className="form-input coupon__input"
              >
                <label
                  className="visually-hidden"
                >
                  Промокод
                </label>
                <input
                  data-test="test-change-promo"
                  id="coupon"
                  name="coupon"
                  onChange={[Function]}
                  placeholder="Введите промокод"
                  type="text"
                />
                <p
                  className="form-input__message form-input__message--success"
                >
                  Промокод принят
                </p>
              </div>
              <button
                className="button button--big coupon__button"
              >
                Применить
              </button>
            </form>
          </div>
          <div
            className="cart__total-info"
          >
            <p
              className="cart__total-item"
            >
              <span
                className="cart__total-value-name"
              >
                Всего:
              </span>
              <span
                className="cart__total-value"
              >
                2 000
                 ₽
              </span>
            </p>
            <p
              className="cart__total-item"
            >
              <span
                className="cart__total-value-name"
              >
                Скидка:
              </span>
              <span
                className="cart__total-value cart__total-value--bonus"
              >
                - 15
                 ₽
              </span>
            </p>
            <p
              className="cart__total-item"
            >
              <span
                className="cart__total-value-name"
              >
                К оплате:
              </span>
              <span
                className="cart__total-value cart__total-value--payment"
              >
                1 985
                 ₽
              </span>
            </p>
            <button
              className="button button--red button--big cart__order-button"
            >
              Оформить заказ
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <div
      className="footer__container container"
    >
      <a
        className="footer__logo logo"
        href="/"
      >
        <img
          alt="Логотип"
          className="logo__img"
          height="70"
          src="/img/svg/logo.svg"
          width="70"
        />
      </a>
      <div
        className="socials footer__socials"
      >
        <ul
          className="socials__list"
        >
          <li
            className="socials-item"
          >
            <a
              aria-label="skype"
              className="socials__link"
              href="https://www.skype.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-skype"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="vsco"
              className="socials__link"
              href="https://www.vsco.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-vsco"
                />
              </svg>
            </a>
          </li>
          <li
            className="socials-item"
          >
            <a
              aria-label="pinterest"
              className="socials__link"
              href="https://www.pinterest.com/"
            >
              <svg
                aria-hidden="true"
                className="socials__icon"
                height="24"
                width="24"
              >
                <use
                  xlinkHref="#icon-pinterest"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <section
        className="footer__nav-section footer__nav-section--info"
      >
        <h2
          className="footer__nav-title"
        >
          О нас
        </h2>
        <p
          className="footer__nav-content footer__nav-content--font-secondary"
        >
          Магазин гитар, музыкальных инструментов и гитарная мастерская 
          <br />
           в Санкт-Петербурге.
          <br />
          <br />
          Все инструменты проверены, отстроены 
          <br />
           и доведены до идеала!
        </p>
      </section>
      <section
        className="footer__nav-section footer__nav-section--links"
      >
        <h2
          className="footer__nav-title"
        >
          Информация
        </h2>
        <ul
          className="footer__nav-list"
        >
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Где купить?
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Блог
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Вопрос - ответ
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Возврат
            </a>
          </li>
          <li
            className="footer__nav-list-item"
          >
            <a
              className="link"
              href="#top"
            >
              Сервис-центры
            </a>
          </li>
        </ul>
      </section>
      <section
        className="footer__nav-section footer__nav-section--contacts"
      >
        <h2
          className="footer__nav-title"
        >
          Контакты
        </h2>
        <p
          className="footer__nav-content"
        >
          г. Санкт-Петербург,
          <br />
           м. Невский проспект, 
          <br />
          ул. Казанская 6.
        </p>
        <div
          className="footer__nav-content"
        >
          <svg
            aria-hidden="true"
            className="footer__icon"
            height="8"
            width="8"
          >
            <use
              xlinkHref="#icon-phone"
            />
          </svg>
          <a
            className="link"
            href="tel:88125005050"
          >
             8-812-500-50-50
          </a>
        </div>
        <p
          className="footer__nav-content"
        >
          Режим работы:
          <br />
          <span
            className="footer__span"
          >
            <svg
              aria-hidden="true"
              className="footer__icon"
              height="13"
              width="13"
            >
              <use
                xlinkHref="#icon-clock"
              />
            </svg>
            <span>
               с 11:00 до 20:00
            </span>
            <span>
              без выходных
            </span>
          </span>
        </p>
      </section>
    </div>
  </footer>
</div>
`;
